contracts/core/exchange-wrappers/KyberNetworkWrapper.sol

```diff
@@ -79,105 +79,107 @@ contract KyberNetworkWrapper is
      *
      *
-     * @param  _maker         Address of issuance order signer to conform to IExchangeWrapper
-     * @param  _              Unused address of fillOrder caller to conform to IExchangeWrapper
-     * @param  _tradeCount    Amount of trades in exchange request
-     * @param  _tradesData    Byte string containing (multiple) Kyber trades
-     * @return address[]      Array of token addresses traded for
-     * @return uint256[]      Array of token amounts traded for
+     * @param  _maker               Address of issuance order signer to conform to IExchangeWrapper
+     * -- Unused address of fillOrder caller to conform to IExchangeWrapper --
+     * @param  _makerToken          Address of maker token used in exchange orders
+     * @param  _makerAssetAmount    Amount of issuance order maker token to use on this exchange
+     * @param  _tradeCount          Amount of trades in exchange request
+     * @param  _tradesData          Byte string containing (multiple) Kyber trades
+     * @return address[]            Array of token addresses traded for
+     * @return uint256[]            Array of token amounts traded for
      */
     function exchange(
         address _maker,
-        address _,
+        address,
+        address _makerToken,
+        uint256 _makerAssetAmount,
         uint256 _tradeCount,
         bytes _tradesData
     )
         external
         onlyAuthorized
         returns (address[], uint256[])
     {
-        address[] memory takerTokens = new address[](_tradeCount);
-        uint256[] memory takerAmounts = new uint256[](_tradeCount);
+        // Ensure the source token is allowed to be transferred by KyberNetworkProxy
+        ERC20.ensureAllowance(
+            _makerToken,
+            address(this),
+            kyberNetworkProxy,
+            _makerAssetAmount
+        );
+
+        address[] memory componentTokensReceived = new address[](_tradeCount);
+        uint256[] memory componentTokensAmounts = new uint256[](_tradeCount);
 
-        uint256 scannedBytes = 0;
+        // Parse and execute the trade at the current offset via the KyberNetworkProxy, each kyber trade is 160 bytes
         for (uint256 i = 0; i < _tradeCount; i++) {
-            // Parse Kyber trade of current offset
-            KyberTrade memory trade = parseKyberTrade(
+            (componentTokensReceived[i], componentTokensAmounts[i]) = tradeOnKyberReserve(
                 _tradesData,
-                scannedBytes
+                i.mul(160)
             );
+        }
 
-            // Execute the trade via the KyberNetworkProxy
-            takerTokens[i] = trade.destinationToken;
-            takerAmounts[i] = tradeOnKyberReserve(
-                trade,
-                _maker
+        // Transfer any unused or remainder maker token back to the issuance order user
+        uint remainderSourceToken = ERC20.balanceOf(_makerToken, this);
+        if (remainderSourceToken > 0) {
```

Comment:
⚠️ If the `componentTokensReceived` during the trade was the `makerToken`, then an unintended amount would be returned to the _maker and the issuance order would fail.

Reply:
adding comment to request changes

Reply:
why is that a problem? why wouldn't we want it to revert? this seems like an optimization to make at the assertion level